pkg/scheduler/core/generic_scheduler.go

```diff
@@ -695,42 +695,47 @@ func pickOneNodeForPreemption(nodesToVictims map[*v1.Node]*Victims) *v1.Node {
 		numPDBViolatingPods := victims.numPDBViolations
 		if numPDBViolatingPods < minNumPDBViolatingPods {
 			minNumPDBViolatingPods = numPDBViolatingPods
-			minPDBViolatingNodes = nil
+			lenNodes1 = 0
```

Comment:
You must set `minNodes1 = nil` as well. Otherwise, `append` at line 701 will keep adding to the existing slice.

Reply:
Updated, PTAL @bsalamat